src/Evolution/DiscontinuousGalerkin/SlopeLimiters/HwenoModifiedSolution.hpp

```diff
@@ -0,0 +1,497 @@
+  }
+
+  // Identify element with maximum mean difference
+  const auto mean_difference = [&tensor_index, &local_mean ](
+      const std::pair<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+                      Package>& neighbor_and_data) noexcept {
+    return fabs(
+        get<::Tags::Mean<Tag>>(neighbor_and_data.second.means)[tensor_index] -
+        local_mean);
+  };
+  const auto max_difference_neighbor_and_data = *alg::max_element(
+      neighbor_data,
+      [&mean_difference](
+          const std::pair<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+                          Package>& lhs,
+          const std::pair<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>,
+                          Package>& rhs) noexcept {
+        return mean_difference(lhs) < mean_difference(rhs);
+      });
+
+  std::vector<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>
+      neighbors_to_exclude{{max_difference_neighbor_and_data.first}};
+
+  // See if other elements share this maximum mean difference. This loop should
+  // only rarely find other neighbors with the same maximal mean difference to
+  // add to the vector, so it will usually not change the vector.
+  const double max_difference =
+      mean_difference(max_difference_neighbor_and_data);
+  for (const auto& neighbor_and_data : neighbor_data) {
+    const auto& neighbor = neighbor_and_data.first;
+    if (neighbor == primary_neighbor or
+        neighbor == max_difference_neighbor_and_data.first) {
+      continue;
+    }
+    const double difference = mean_difference(neighbor_and_data);
+    if (UNLIKELY(equal_within_roundoff(difference, max_difference))) {
+      neighbors_to_exclude.push_back(neighbor);
+    }
+  }
+
+  return neighbors_to_exclude;
+}
+
+// Compute the vector b_s for the constrained fit. For details, see the
+// documentation of `compute_hweno_modified_neighbor_solution` below.
+template <typename Tag, size_t VolumeDim, typename Package>
+DataVector b_vector(
+    const Mesh<VolumeDim>& mesh, const size_t tensor_index,
+    const DataVector& quadrature_weights,
+    const DirectionMap<VolumeDim, Matrix>& interpolation_matrices,
+    const DirectionMap<VolumeDim, DataVector>&
+        quadrature_weights_dot_interpolation_matrices,
+    const std::unordered_map<
+        std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, Package,
+        boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+        neighbor_data,
+    const std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>&
+        primary_neighbor,
+    const std::vector<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>&
+        neighbors_to_exclude) noexcept {
+  const size_t number_of_grid_points = mesh.number_of_grid_points();
+  DataVector b(number_of_grid_points, 0.);
+
+  for (const auto& neighbor_and_data : neighbor_data) {
+    if (alg::found(neighbors_to_exclude, neighbor_and_data.first)) {
```

Comment:
For typical invocations in 3D, this `if` should return true ~ 1/6th of the time. As I understood things, this is not unlikely enough to justify an `UNLIKELY`...?

Reply:
Meh :man_shrugging: In general the compiler just randomly chooses one of the branches to be the `LIKELY` branch, so giving it the info isn't bad. Then again it almost certainly won't be the expensive thing. Having said that, getting code instruction cache hits is surprisingly important. I would ultimately say "Yes, it would be good to help the compiler"